### DISCUSSION TITLE
OpenCV for timelapse?

### DISCUSSION THREAD

**Discussion by DominicWeston** [xblax/flashforge_ad5m_klipper_mod#139] [Ideas]
Several years ago I played with a program someone wrote for the RPi  that used OpenCV to calculate the speed of a car by timing when it enters a point on a frame and times how long it took to reach another point in a later frame. So, could OpenCV running on my laptop or rpi track the video stream for some change that tells it to save that frame, like the bed moving down or ??? This would offload all image handling to somewhere other than the printers processor. Just a thought.

**Comment by xblax** [comment]
No, not really a good idea. The printer already knows when the layer changes and just saving a frame form the webcam is not much work for it. Rendering the final timelapse video could be offloaded, but I think we will manage to do that on the printer as well.

**Reply by DominicWeston** [reply]
Thanks
